privacy_analytics/managers.py

```diff
@@ -0,0 +1,25 @@
+from django.db.models import Manager, Avg, Count, Q, FloatField
+from django.db.models.functions import Cast
+
+
+class PageViewManager(Manager):
+    def unique(self, **kwargs):
+        return self.filter(**kwargs).values('session_key').distinct().count()
+
+    def views_per_visitor(self, **kwargs):
+        return self.filter(**kwargs).values('session_key').annotate(total=Count('pk')).aggregate(Avg('total'))
+
+    def total_views(self, **kwargs):
+        return self.filter(**kwargs).values('session_key').count()
+
+    def percent_authenticated(self, **kwargs):
+        return self.filter(**kwargs).aggregate(total=(Cast(Count('is_authenticated', filter=Q(is_authenticated=True)), FloatField()) / Cast(Count('pk'), FloatField())) * 100.00)
+    
+    def referrers(self, **kwargs):
+        return self.filter(**kwargs).values('referrer').annotate(total=Count('pk')).all()[:10]
```

Comment:
* Does this include the referrers for _all_ page loads, or just initial page loads? I don't remember this codebase well enough to remember if internal referrer handling is handled elsewhere.
* Does this return the top 10 referrers, or does it use some other ordering? Asking mostly because I don't see any sort specified in the db lookup.
* Should we consider adding an option to list more than 10 referrers, maybe as an optional argument? Alternatively, should we just return all of the referrers, and do the slicing elsewhere (in the view, for example)?

Reply:
So I have refactored the `managers.py` and have tidied it up. If there are zero records when it does the math, Django handles that internally and creates no issues.

Right now, on the initial page load for the dashboard, it just loads all records from the database. If we implement pagination, which Django supports, we definitely could add in adding the option to limit!